src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -242,6 +242,107 @@
     [n.strip().upper() for n in snowflake_reserved_keywords.split("\n")]
 )
 
+# Add datetime units and their aliases from
+# https://docs.snowflake.com/en/sql-reference/functions-date-time.html#label-supported-date-time-parts
+snowflake_dialect.sets("datetime_units").clear()
+snowflake_dialect.sets("datetime_units").update(
```

Comment:
If we think this is too long for `dialect_snowflake.py` we could move this list to `dialect_snowflake_keywords.py`?